### PR TITLE
xds: Cleanup udpa -> xds refs

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: xds: Cleanup udpa -> xds refs
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: examples/dynamic-config-cp/Dockerfile-control-plane [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM golang
 
 RUN apt-get -y update && apt-get install -y -qq --no-install-recommends netcat
-RUN git clone https://github.com/envoyproxy/go-control-plane
+RUN git clone -b udpa-xds https://github.com/phlax/go-control-plane
```
Comment (phlax) [inline examples/dynamic-config-cp/Dockerfile-control-plane:4]: this wont be necessary if https://github.com/envoyproxy/go-control-plane/pull/404 lands

File: api/API_VERSIONING.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # API versioning guidelines
 
-The Envoy project (and in the future [UDPA](https://github.com/cncf/udpa)) takes API stability and
+The Envoy project (and in the future [UDPA](https://github.com/cncf/xds)) takes API stability and
```
Comment (phlax) [inline api/API_VERSIONING.md:3]: `[XDS]`  ?

File: GOVERNANCE.md [hunk 3]
```diff
@@ -169,7 +169,7 @@ take on API shepherd review load and participate in meetings. They should be act
 and responsive to GitHub issues and PRs on which they are tagged.
 
 The API shepherds are distinct to the [UDPA working
-group](https://github.com/cncf/udpa/blob/main/README.md), which aims to evolve xDS directionally
```
Comment (htuch) [inline GOVERNANCE.md:172]: Rename to "xDS working group"

File: api/API_VERSIONING.md (hunk 2 above)
Comment (htuch) [reply api/API_VERSIONING.md:3]: `The Envoy project (and [xDS working group])`

File: examples/dynamic-config-cp/Dockerfile-control-plane (hunk 1 above)
Comment (htuch) [reply examples/dynamic-config-cp/Dockerfile-control-plane:4]: Merged

File: GOVERNANCE.md (hunk 3 above)
Comment (htuch) [reply GOVERNANCE.md:172]: Can you also replace other instances of "UDPA" with "xDS" in this file? Ta.

**Comment by phlax** [comment]
not sure whether we need to update go.sum in `ext_authz` (or other examples that use go-control-plane)

im guessing its pegged to a particular version of go-control-plane so its related to that and should work for now

that got me also to thinking to search google - and i noticed that there are some redhat pkgs for `cncf-udpa-*` amongst other things

File: GOVERNANCE.md [hunk 4]
```diff
@@ -168,14 +168,14 @@ developments and expect to be active in this space in the near future. API sheph
...
 
-The API shepherds are distinct to the [UDPA working
-group](https://github.com/cncf/udpa/blob/main/README.md), which aims to evolve xDS directionally
+The API shepherds are distinct to the [xDS working
+group](https://github.com/cncf/xds/blob/main/README.md), which aims to evolve xDS directionally
 towards a universal dataplane API. API shepherds are responsible for the execution of the xDS
-day-to-day and guiding xDS implementation changes. Proposals from UDPA-WG will be aligned with the
-xDS API shepherds to ensure that xDS is heading towards the UDPA goal. xDS API shepherds operate
+day-to-day and guiding xDS implementation changes. Proposals from xDS-WG will be aligned with the
+xDS API shepherds to ensure that xDS is heading towards the xDS goal. xDS API shepherds operate
```
Comment (htuch) [inline GOVERNANCE.md:175]:
```suggestion
xDS API shepherds to ensure that xDS is heading towards the goal of client and server neutral xDS. xDS API shepherds operate
```

Comment (phlax) [reply GOVERNANCE.md]: updated

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**